### ISSUE TITLE
Add methods to read and write strings as CArrays in compressed format

### ISSUE THREAD

**Issue by jason-s** [PyTables/PyTables#631]
I've been using the following kind of approach for a few years to store strings as CArrays in compressed format (and furthermore to contain YAML or JSON for heterogeneous structured non-searchable storage); would it make sense to add something like this into the PyTables API?

```
def recordStringInHDF5(group, nodename, s, complevel=5, complib='zlib'):
    '''creates a CArray object in an HDF5 file
    that represents a unicode string'''

    barray = np.fromstring(s.encode('utf-8'),np.uint8)
    atom = pt.UInt8Atom()
    filters = pt.Filters(complevel=complevel, complib=complib)
    ca = group._v_file.create_carray(group, nodename, atom, shape=(len(barray),),
                               filters=filters)
    ca[:] = barray
    return ca

def retrieveStringFromHDF5(node):
    return unicode(node.read().tostring(), 'utf-8')
```

**Comment by FrancescAlted** [comment]
Yes, that's interesting.  However, I suppose that the 'canonical' solution would be to add something like:
```
UnicodeAtom(itemsize, shape, encoding='utf-8')
```
and this would allow to store arbitrary Unicode strings in HDF5.

However, the above proposal is more work than it seems.

Meanwhile, I think your functions would be a good candidate to be put into a new `utils` subpackage.  Interested in proposing a PR?

**Comment by jason-s** [comment]
>Interested in proposing a PR?

Yeah, it may take a while though. I'm good with snippets, not really sure how best to organize them or unit tests or things like that which constitute good assimilation into the pytables library.
